src/action-ros-ci.ts

```diff
@@ -225,16 +226,66 @@ async function run() {
 		await io.rmRF(rosWorkspaceDir);
 
 		// Checkout ROS 2 from source and install ROS 2 system dependencies
-		await io.mkdirP(rosWorkspaceDir + "/src");
+		await io.mkdirP(path.join(rosWorkspaceDir, "src"));
+
+		// Download repos files into a specific directory
+		const reposFilesDir = path.join(rosWorkspaceDir, "repos-files");
+		await io.mkdirP(reposFilesDir);
+		const curlFlags = curlFlagsArray.join(" ");
+		const reposFiles: string[] = [];
+		for (const [
+			index,
+			vcsRepoFileUrl,
+		] of vcsRepoFileUrlListResolved.entries()) {
+			const reposFile = `${index}.repos`;
+			await execBashCommand(
+				`curl ${curlFlags} '${vcsRepoFileUrl}' --output ${reposFile}`,
+				undefined,
+				{ cwd: reposFilesDir }
+			);
+			reposFiles.push(path.join(reposFilesDir, reposFile));
+		}
 
 		const options = {
 			cwd: rosWorkspaceDir,
 		};
 
-		const curlFlags = curlFlagsArray.join(" ");
-		for (const vcsRepoFileUrl of vcsRepoFileUrlListResolved) {
+		// Process repos files and apply PR dependency changes
+		const prDependencies = dep.getPrDependencies(github.context.payload);
+		if (prDependencies.length === 0) {
+			await core.group("No PR dependencies", () => Promise.resolve());
+		} else {
+			await core.group("PR dependencies", () => {
+				for (const dependency of prDependencies) {
+					core.info("\t" + dep.getFullUrlFromDependency(dependency));
+				}
+				return Promise.resolve();
+			});
```

Comment:
I think I'd like to avoid too much deeply nested code in the `else` block - and this logging could be like follows. The remainder doesn't need to be conditional, as long as it can handle an empty list.

```suggestion
                await core.group(`PR dependencies${prDependencies.length ? '' : ' - None Detected!', () => {
                        for (const dependency of prDependencies) {
				core.info("\t" + dep.getFullUrlFromDependency(dependency));
			}
			return Promise.resolve();
		}
```

Reply:
Sure, done in 0fb6367d22e27a6c36e3f1896acb9d5c2d4c87ed